static/app/views/performance/traceDetails/utils.tsx

```diff
@@ -68,7 +72,8 @@ export function getTraceDetailsUrl({
       query: {
         ...queryParams,
         timestamp: getTimeStampFromTableDateField(timestamp),
-        eventId,
+        eventId: targetId ?? eventId,
```

Comment:
You're already passing the `targetId` through so we should prefer the `eventId` here in case it's set.
```suggestion
        eventId,
```